src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -6302,6 +6302,14 @@ new_mmap(void *addr, size_t length, int prot, int flags, int fd, off_t offset)
 
 	atomic_fetch_add_relaxed(&num_mmap, 1);
 
+	if ((fd < FD_FILE_BASE) && (fd_directed >= FD_FILE_BASE) && d_compatible_mode) {
+		/* DAOS-14494: Force the kernel to update the size before mapping. */
```

Comment:
I've linked DAOS-14494 and DAOS-16556 in Jira.

Reply:
Sounds good. Thank you very much!